### DISCUSSION TITLE
V10 - Best way to use custom png on SymbolLayer

### DISCUSSION THREAD

**Discussion by FrancoisVA** [rnmapbox/maps#2249] [Q&A]
Hi, I'm looking for the best way to use custom png's for symbol layers. I used to be able to just require a local image and use in in the style prop as iconImage. Like this:

```
const PIN_DROP = require('../images/pin.png');

const pinDropStyle = {
  iconImage: PIN_DROP,
  iconIgnorePlacement: true,
  iconAllowOverlap: true,
};

<MapboxGL.ShapeSource
  onPress={e => onShapeSourceSavedUserGeneratedLayer(e)}
  id={`SUGL_SourceId${Uuid}.${index}`}
  key={`SUGL_SourceKey${Uuid}.${index}`}
  shape={GeoJson}>
      <MapboxGL.SymbolLayer
        id={`SUGL_SymbolLayerId${Uuid}.${index}`}
        key={`SUGL_SymbolLayerKey${Uuid}.${index}`}
        maxZoomLevel={AppStore.zoomLevels.PIN_DROP_MAX_ZOOM}
        minZoomLevel={AppStore.zoomLevels.PIN_DROP_MIN_ZOOM}
        style={pinDropStyle}
      />
</MapboxGL.ShapeSource>
```
  
When I use this method now, my app crashes so I'm guessing that this is not the right way to do this anymore.

**Comment by KiwiKilian** [comment]
@FrancoisVA can you add some more code, how you apply this to the `SymbolLayer`?

**Reply by FrancoisVA** [reply]
Hi, sorry for the late reply, I edited my question. 
I assign that style to the Symbol layer, the Symbol layer is the child of a shapeSource.

**Reply by KiwiKilian** [reply]
Does it give any error in the logs?

Compare your code with [this example](https://github.com/rnmapbox/maps/blob/main/example/src/examples/SymbolCircleLayer/CustomIcon.js#L14). Try `import` or removing `iconIgnorePlacement`. Otherwise a minimal example, base on the [BugReportTemplate](https://github.com/rnmapbox/maps/blob/main/example/src/examples/BugReportExampleTS.tsx) would be helpful.
